src/codecs/dds.rs

```diff
@@ -71,6 +71,16 @@ struct Header {
     caps2: u32,
 }
 
+/// Extended DX10 header used by some DDS image files
+#[derive(Debug)]
+struct DX10Header {
+    dxgi_format: u32,
+    resource_dimension: u32,
+    misc_flag: u32,
+    array_size: u32,
+    misc_flags_2: u32,
+}
```

Comment:
Is there any of these fields that should require some form of validation? If not, why? They are being read but disregarded which is not very forward compatible and could be a slight hazard if that causes parsing result to change unexpectedly in the future.

Reply:
Should I add the validation for the fields in the `from_reader` function?

Reply:
A separate `fn validate(&self) -> Result<()>` should suffice. At time we'll likely want to change it to return some kind of parsing result (i.e. feed it to the color type/metadata recorder) and then it's a bit easier if it's separate from reading.